api/wasm/cpp/proxy_wasm_api.h

```diff
@@ -522,6 +523,32 @@ inline WasmResult getPluginDirection(PluginDirection *direction_ptr) {
   PROXY_EXPRESSION_GET_UINT32("plugin.direction", reinterpret_cast<uint32_t*>(direction_ptr));
 }
 
+// Generic selector
+inline absl::optional<WasmDataPtr> getSelectorExpression(std::initializer_list<absl::string_view> parts) {
```

Comment:
I thought we had decided on using the same representation that Pairs uses:
numberOfPathElements : uint32
pathElement0Size : uint32
pathElement1Size : uint32
...
pathElement0 : char[pathElement0Size]
pathElement1 : char[pathElement1Size]
...

This would be very fast to convert into a vector of string_view:

std::vector<string_view> path;
auto size = *reinterpret_cast<uint32_t*>(&buf[0]);
path.reserve(size);
for (int i = 0; i < size; i++) {
  path.push_back(string_view( ....
}

This encoding requires searching for the nulls, although that is not all that expensive since it can be done in 64-bit chunks for large path parts.

I am OK with this, it just isn't what I was expecting.

Reply:
Yes, sorry. I was trying to make it allocation-free, and this is what it ended up with. I think it's nice to have a streaming-like access since it matches what the code does in the end. I had a vector at first, but then it looked like it wasn't necessary.

Reply:
you can still do it allocation free with the above data structure, but I agree that this isn't bad.  let's keep it